src/components/foodDescription/FoodDescription.js

```diff
@@ -1,44 +1,159 @@
 // React
-import { useMemo, useRef } from "react";
+import { useMemo, useRef, useState } from "react";
 
 // Libraries
+import React from "react";
 import PropTypes from "prop-types";
 
 // lodash
 import _keys from "lodash/keys";
 import _toLower from "lodash/toLower";
+import _has from "lodash/has";
+import _omit from "lodash/omit";
+import _isEmpty from "lodash/isEmpty";
 import _find from "lodash/find";
 
 // components
 import Sidebar from "./components/sidebar";
 import FoodList from "./components/foodList";
+import Cart from "../cart";
+import Icon from "../../commonComponents/icon";
+
+// helpers
+import {
+    getCartItems,
+    getTotalFromCartItems,
+} from "./helpers/foodDescription.general";
+
+// constants
+import {
+    CART_ITEMS,
+    EMPTY_CART_DESCRIPTION,
+} from "./constants/foodDescription.general";
+
+// utility
+import saveToLocalStorage from "../../utility/saveToLocalStorage";
+
+// icons
+import emptyCartImage from "../../assets/images/cartEmpty.webp";
 
 // css
 import "./foodDescription.css";
 
 function FoodDescription(props) {
     const { foodList } = props;
-    const keys = useMemo(() => _keys(foodList), [foodList]);
+
+    const categories = useMemo(() => _keys(foodList), [foodList]);
     const foodListSectionRefs = useRef([]);
 
+    const [cart, setCart] = useState({});
+
     function setRef(ele) {
         if (!foodListSectionRefs.current.includes(ele)) {
             foodListSectionRefs.current.push(ele);
         }
     }
 
     function handleCategoryScroll(category) {
-        const section = _find(foodListSectionRefs.current, { id: _toLower(category) });
+        const section = _find(foodListSectionRefs.current, {
+            id: _toLower(category),
+        });
         section.scrollIntoView({ behavior: "smooth" });
     }
 
+    function handleAddToCart(event) {
+        const {
+            target: {
+                dataset: { id },
+            },
+        } = event;
+        const isIdPresentInCart = _has(cart, id);
+        if (!isIdPresentInCart) {
+            setCart({ ...cart, [id]: 1 });
+        }
+    }
+
+    function handleIncrementClick(event) {
+        const {
+            target: {
+                parentNode: {
+                    dataset: { id },
+                },
+            },
+        } = event;
+        const newQuantitiy = cart[id] + 1;
```

Comment:
handle these logic in a helper and that helper will return a new state and update here